### PR TITLE
Add support for labels and taints to HostedCP NodePools

### PR DESCRIPTION
**Pull request by andreadecorte**
This covers both creation and modification. The UX should be the same as Classic ROSA.

Related: [SDA-8219](https://issues.redhat.com//browse/SDA-8219)

### CODE REVIEW

**Comment by andreadecorte** [comment]
Draft pending https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/merge_requests/5313

File: cmd/create/machinepool/nodepool.go [hunk 1]
```diff
@@ -139,8 +139,51 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
...
+	}
+	taintBuilders, err := parseTaints(taints)
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
 	npBuilder := cmv1.NewNodePool()
-	npBuilder.ID(name)
+	npBuilder.ID(name).Labels(labelMap).
```
Comment (gdbranco) [inline cmd/create/machinepool/nodepool.go:185]: Does it not needs npBuilder = ...?

File: cmd/create/machinepool/nodepool.go [hunk 2]
```diff
@@ -139,8 +139,51 @@ func addNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r
...
+			r.Reporter.Errorf("Expected a valid comma-separated list of attributes: %s", err)
+			os.Exit(1)
+		}
+	}
+	taintBuilders, err := parseTaints(taints)
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
```
Comment (zgalor) [inline cmd/create/machinepool/nodepool.go:183]: surely this code can be exported to functions used both by machine pools and node pools.
There is nothing nodepool-specific here

File: cmd/create/machinepool/nodepool.go (hunk 1 above)
Comment (andreadecorte) [reply cmd/create/machinepool/nodepool.go:185]: no, not needed from my understanding and tests

File: cmd/create/machinepool/nodepool.go (hunk 2 above)
Comment (andreadecorte) [reply cmd/create/machinepool/nodepool.go:183]: yes makes sense, as I did it already on the edit part

Comment (andreadecorte) [reply cmd/create/machinepool/nodepool.go:145]: refactored

**Comment by zgalor** [comment]
lgtm.
Did you check machine pools labels & taint flows weren't broken?

**Comment by andreadecorte** [comment]
> lgtm. Did you check machine pools labels & taint flows weren't broken?
not yet. let me check with a classic cluster before proceeding with a merge

**Comment by andreadecorte** [comment]
@zgalor checked a non-hypershift cluster, working as expected

**Pull request merged**